fuse_models/include/fuse_models/unicycle_2d_state_cost_function.h

```diff
@@ -147,8 +147,12 @@ class Unicycle2DStateCostFunction : public ceres::SizedCostFunction<8, 2, 1, 2,
       acc_linear_pred_y,
       jacobians);
 
-    residuals[0] = parameters[5][0] - position_pred_x;
-    residuals[1] = parameters[5][1] - position_pred_y;
+    // rotate the position residual into the local frame
+    auto sin_pred_inv = std::sin(-yaw_pred);
+    auto cos_pred_inv= std::cos(-yaw_pred);
+
+    residuals[0] = cos_pred_inv * delta_position_pred_x - sin_pred_inv * delta_position_pred_y;
+    residuals[1] = sin_pred_inv * delta_position_pred_x + cos_pred_inv * delta_position_pred_y;
     residuals[2] = parameters[6][0] - yaw_pred;
```

Comment:
I don't see why `residuals[0]` and `residual[1]` no longer depend on the previous pose, defined in `parameters[0]` and `parameters[1]`, i.e. `position1_x`, `position1_y` and `yaw1`, and the next pose, defined in `parameters[5]`, i.e. `position2_x` and `position2_y`.

Shouldn't we be computing the delta between those two and them compute the residual as the delta between that and the predicted pose. That is:
```suggestion
    const Eigen::Matrix<T, 2, 1> position1(parameters[0][0], parameters[0][1]);
    const Eigen::Matrix<T, 2, 1> position2(parameters[5][0], parameters[5][1]);

    const Eigen::Matrix<T, 2, 1> delta_position = fuse_core::rotationMatrix2D(parameters[1][0]).transpose() * (position2 - position1);
    const T delta_yaw = parameters[6][0] - parameters[1][0];  // omitting fuse_core::wrapAngle2D because it is done later on

    const Eigen::Matrix<T, 2, 1> delta_position_pred(delta_position_pred_x, delta_position_pred_y);

    residuals.template head<2>() = fuse_core::rotationMatrix2D(delta_yaw).transpose() * (delta_position_pred - delta_position);
    residuals[2] = delta_yaw - yaw_pred;  // omitting fuse_core::wrapAngle2D because it is done later on
```

In a way, this is similar to https://github.com/locusrobotics/fuse/blob/2c652c7450c24f9fe462de0c33d5c2c16e21c37d/fuse_constraints/include/fuse_constraints/normal_delta_pose_2d_cost_functor.h#L118-L125

Reply:
In this case, I feel it would just be better to keep what I originally had, and just rotate the error back into the local frame